kratos/solving_strategies/strategies/solving_strategy.h

```diff
@@ -29,426 +29,493 @@
+
+            noalias(itNode->Coordinates()) = itNode->GetInitialPosition().Coordinates();
+            noalias(itNode->Coordinates()) += itNode->FastGetSolutionStepValue(DISPLACEMENT);
+        }
+        
+        if (this->GetEchoLevel() != 0 && GetModelPart().GetCommunicator().MyPID() == 0 )
+        {
+            std::cout<<" MESH MOVED "<<std::endl;
+        }
+
+        KRATOS_CATCH("");
+    }
 
+    /**
+     * Operations to get the pointer to the model
+     * @return mrModelPart: The model part member variable
+     */
     inline ModelPart& GetModelPart()
     {
-        return mr_model_part;
+        return mrModelPart;
     };
 
+    /**
+     * Operations to get the residual norm
+     * @return The residual norm
+     */
     virtual double GetResidualNorm()
     {
+        KRATOS_ERROR << "You are calling to the base class method GetResidualNorm, you are evil, and your seed must be eradicated from the face of the earth" << std::endl;
         return 0.0;
     }
 
     /**
-     * function to perform expensive checks.
+     * Function to perform expensive checks.
      * It is designed to be called ONCE to verify that the input is correct.
      */
     virtual int Check()
     {
-        KRATOS_TRY
+        KRATOS_TRY;
 
-        //check if displacement var is needed
+        // Check if displacement var is needed
         if (mMoveMeshFlag == true)
         {
-            for (ModelPart::NodeIterator i = GetModelPart().NodesBegin();
-                    i != GetModelPart().NodesEnd(); ++i)
-
-                if (i->SolutionStepsDataHas(DISPLACEMENT) == false)
+            NodesArrayType& NodesArray = GetModelPart().Nodes();
+            int numNodes = static_cast<int>(NodesArray.size());
+
+//             #pragma omp parallel for
+            for(int i = 0; i < numNodes; i++)  
+            {
+                auto itNode = NodesArray.begin() + i;
+                
+                if (itNode->SolutionStepsDataHas(DISPLACEMENT) == false)
                 {
-                    std::cout << "problem on node with Id " << i->Id() << std::endl;
-                    KRATOS_THROW_ERROR(std::logic_error, "It is impossible to move the mesh since the DISPLACEMENT var is not in the model_part. Either use SetMoveMeshFlag(False) or add DISPLACEMENT to the list of variables", "");
+                    std::cout << "problem on node with Id " << itNode->Id() << std::endl;
+                    KRATOS_ERROR << "It is impossible to move the mesh since the DISPLACEMENT var is not in the rModelPart. Either use SetMoveMeshFlag(False) or add DISPLACEMENT to the list of variables" << std::endl;
                 }
+            }
         }
 
+        ElementsArrayType& ElementsArray = GetModelPart().Elements();
+        int numElements = static_cast<int>(ElementsArray.size());
 
-
-        for (ModelPart::ElementsContainerType::iterator it = GetModelPart().ElementsBegin();
-                it != GetModelPart().ElementsEnd(); it++)
```

Comment:
this code was perfectly valid, and it was intended not to run in parallel. Why did u change it? (same for nodes and conditions)